File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/course/CourseService.java
package kh.farrukh.sortmapperexamplejava.course;

import java.util.List;
import kh.farrukh.sortmapperexamplejava.course.model.CourseCreateRequestDTO;
import kh.farrukh.sortmapperexamplejava.course.model.CourseDetailsResponseDTO;
import kh.farrukh.sortmapperexamplejava.course.model.CourseResponseDTO;
import kh.farrukh.sortmapperexamplejava.course.model.CourseShortInfoResponseDTO;
import kh.farrukh.sortmapperexamplejava.course.model.CourseUpdateRequestDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;

public interface CourseService {

  void createCourse(CourseCreateRequestDTO requestDTO);

  Page<CourseResponseDTO> getCourses(Pageable pageable);

  List<CourseShortInfoResponseDTO> getCoursesShortInfo(Sort sort);

  CourseDetailsResponseDTO getCourse(Long id);

  void updateCourse(Long id, CourseUpdateRequestDTO requestDTO);

  void deleteCourse(Long id);

  CourseEntity findCourse(Long id);

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/course/CourseService.kt
package kh.farrukh.sortmapperexample.course

import kh.farrukh.sortmapperexample.course.model.*
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort

interface CourseService {

    fun getCourseById(id: Long): CourseDetailsResponseDTO

    fun getCourses(pageable: Pageable): Page<CourseResponseDTO>

    fun createCourse(requestDTO: CourseCreateRequestDTO)

    fun updateCourse(id: Long, requestDTO: CourseUpdateRequestDTO)

    fun deleteCourse(id: Long)

    fun findCourse(id: Long): CourseEntity

    fun getCoursesShortInfo(sort: Sort): List<CourseShortInfoResponseDTO>
}